src/ucm/cuda/cudamem.c

```diff
@@ -99,6 +99,34 @@ ucm_dispatch_mem_type_free(void *addr, size_t length, ucs_memory_type_t mem_type
     ucm_event_dispatch(UCM_EVENT_MEM_TYPE_FREE, &event);
 }
 
+static ucs_memory_type_t ucm_cuda_get_mem_type(void *ptr)
+{
+    struct cudaPointerAttributes ptr_attrs;
+    ucs_memory_type_t mem_type;
+    cudaError_t cu_err;
+
+    cu_err = cudaPointerGetAttributes(&ptr_attrs, ptr);
+    if (cu_err != cudaSuccess) {
+        ucm_warn("cudaPointerGetAttributes(ptr=%p) failed: %s",
+                 ptr, cudaGetErrorString(cu_err));
+        return UCS_MEMORY_TYPE_CUDA;
+    }
+
+#if defined(HAVE_STRUCT_CUDAPOINTERATTRIBUTES_TYPE)
+    mem_type = ((ptr_attrs.type == cudaMemoryTypeManaged) ?               
+                UCS_MEMORY_TYPE_CUDA_MANAGED :
+                UCS_MEMORY_TYPE_CUDA);
+#elif defined(HAVE_STRUCT_CUDAPOINTERATTRIBUTES_ISMANAGED)
+    mem_type = (ptr_attrs.isManaged ?
```

Comment:
btw, `isManaged` should be always present, right?
For instance `uct_cuda_base_detect_memory_type` checks just it (after ensuring that mem_type isDevice). Why do we need to check `memtype`?

Reply:
btw, we may just use the same approach as we have in `uct_cuda_base_detect_memory_type`, wdyt?